src/magic/model/MagicPlayer.java

```diff
@@ -483,7 +486,12 @@ void createHandAndLibrary(final int handSize) {
             final MagicCardDefinition cardDefinition = deck.get(i);
             if (cardDefinition.isValid()) {
                 final long id = currGame.getUniqueId();
-                library.add(new MagicCard(cardDefinition,this,id));
+                try {
+                    library.add(new MagicCard(cardDefinition, this, id));
```

Comment:
The exception should be caught closer to the place where it is thrown, in this case it is thrown by the line to loadAbilities in the MagicCard constructor.

Reply:
caught and rethrown now in  the MagicCard constructor.